### DISCUSSION TITLE
RuntimeError: asyncio.run() cannot be called from a running event loop

### DISCUSSION THREAD

**Discussion by michelebon** [safevideo/autollm#214] [Q&A] [answered]
Hi everybody, i'm new to python and llm world. I'm following some tutorials online to improve my knowledge. Since two days this tutorial gives this error in the first stage of "Advanced Usage" step:

2024-01-17 21:17:17,191 - autollm - WARNING - use_async is only supported for SimpleVectorStore. Ignoring use_async.
Parsing nodes: 100%
1/1 [00:04<00:00, 4.53s/it]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
[<ipython-input-5-d716a46987fb>](https://localhost:8080/#) in <cell line: 36>()
     34 similarity_top_k = 3
     35 
---> 36 query_engine = AutoQueryEngine.from_defaults(
     37     documents=documents,
     38     llm_model=llm_model,

10 frames
[/usr/lib/python3.10/asyncio/runners.py](https://localhost:8080/#) in run(main, debug)
     31     """
     32     if events._get_running_loop() is not None:
---> 33         raise RuntimeError(
     34             "asyncio.run() cannot be called from a running event loop")
     35 

RuntimeError: asyncio.run() cannot be called from a running event loop

How can I solve it? Thanks for help.

**Comment by michelebon** [comment]
i'm sorry, I didn't link the tutorial. 

https://colab.research.google.com/github/safevideo/autollm/blob/main/examples/quickstart.ipynb

**Comment by talhasarit** [comment, marked as answer]
Hi @michelebon ,

Thank you so much for exploring autollm and for bringing this issue to our attention! We're thrilled to have you in our community. To quickly resolve the error you're encountering, you can use the `nest_asyncio` package. This package allows for nesting the asyncio event loop, which should fix the `RuntimeError` you're experiencing.

Here's what you can do:

1. Right before the line where the error occurs, add the following code:

    ```python
    import nest_asyncio
    nest_asyncio.apply()
    ```

2. Additionally, change the `embed_model` parameter to `embed_model="text-embedding-ada-002",`.

This should help you move past the error. Meanwhile, I will be working on fixing this and updating the quickstart Colab notebook to address this issue more permanently. Keep experimenting and let us know if you encounter any more challenges!
